mypyc/ops.py

```diff
@@ -1696,6 +1885,94 @@ def concrete_subclasses(self) -> List['ClassIR']:
         # to get stable order.
         return sorted(concrete, key=lambda c: (len(c.children), c.name))
 
+    def serialize(self) -> JsonDict:
+        return {
+            'name': self.name,
+            'module_name': self.module_name,
+            'is_trait': self.is_trait,
+            'is_ext_class': self.is_ext_class,
+            'is_abstract': self.is_abstract,
+            'is_generated': self.is_generated,
+            'inherits_python': self.inherits_python,
+            'has_dict': self.has_dict,
+            'builtin_base': self.builtin_base,
+            'ctor': self.ctor.serialize(),
+            # We serialize dicts as lists to ensure order is preserved
+            'attributes': [(k, t.serialize()) for k, t in self.attributes.items()],
+            # We serialize decls out if the decl isn't in methods otherwise
+            # we refer to the whatever
```

Comment:
Something seems to be missing in the comment: "... to the whatever". Also, this kind of explains what the code does -- maybe this can be written in a more abstract or purpose oriented way?

Also, it would be good to document `methods` in `__init__`.

Reply:
... In grad school I once accidentally submitted a fellowship application that listed "AND STUFF" as one of my skills.